--- venv/lib/python3.8/site-packages/montreal_forced_aligner/abc.py	2022-04-28 19:09:22.000000000 +0100
+++ abc.py	2022-04-29 10:21:59.031221800 +0100
@@ -194,6 +194,7 @@
         debug: bool = False,
         verbose: bool = False,
         quiet: bool = False,
+        logger: Any = logging.getLogger('dummy'),
         **kwargs,
     ):
         super().__init__(**kwargs)
@@ -202,6 +203,7 @@
         self.use_mp = use_mp
         self.dirty = False
         self.quiet = quiet
+        self.logger = logger
 
     def log_debug(self, message: str = "") -> None:
         """
